pkg/validation/commands.go

```diff
@@ -130,11 +130,16 @@ func validateCommandComponent(command v1alpha2.Command, components []v1alpha2.Co
 
 	// must map to a container component
 	for _, component := range components {
-		if component.Container != nil && commandComponent == component.Name {
-			return nil
+		if commandComponent == component.Name {
+			if component.Container != nil {
+				return nil
+			}
+			if command.Apply != nil && (component.Image != nil || component.Kubernetes != nil || component.Openshift != nil) {
+				return nil
+			}
```

Comment:
I would break after this line. If none of the conditions are satisfied where there is a name match, then no need to iterate further